core/src/main/java/nu/yona/server/subscriptions/service/UserService.java

```diff
@@ -76,22 +83,25 @@ public UserDTO addUser(UserDTO userResource)
 		return UserDTO.createInstanceWithPrivateData(userEntity);
 	}
 
+	@Autowired
+	UserServiceTempEncryptionContextExecutor tempEncryptionContextExecutor;
+
+	public User addUserCreatedOnBuddyRequest(UserDTO buddyUserResource, String tempPassword)
+	{
+		UUID savedUserID = CryptoSession.execute(Optional.of(tempPassword), null, () -> tempEncryptionContextExecutor
+				.addUserCreatedOnBuddyRequestFlush(buddyUserResource).getID());
+		return getEntityByID(savedUserID);
+	}
+
 	@Transactional
 	public UserDTO updateUser(UUID id, UserDTO userResource)
 	{
 		User originalUserEntity = getEntityByID(id);
-		UserDTO savedUser;
 		if (originalUserEntity.isCreatedOnBuddyRequest())
```

Comment:
Notice the app (or any other hacker) can arrive here if it uses the temp password as Yona-Password and could use the temp password forever...
So it might be a security issue: if someone intercepts the email, she may update the user without storing a new password. Maybe not an issue.